lib/srv/heartbeatv2.go

```diff
@@ -73,16 +108,15 @@ func NewSSHServerHeartbeat(cfg SSHServerHeartbeatConfig) (*HeartbeatV2, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	fetcher := newMetadataFetcher()
 	inner := &sshServerHeartbeatV2{
 		getServer: func(ctx context.Context) *types.ServerV2 {
 			server := cfg.GetServer()
-			metadata, err := metadata.Get(ctx)
-			if err == nil {
-				if metadata.CloudMetadata != nil {
-					server.SetCloudMetadata(metadata.CloudMetadata)
-				}
-			} else {
+			metadata, err := fetcher.get(ctx)
+			if err != nil {
 				log.Warnf("Failed to get metadata: %v", err)
+			} else if metadata != nil && metadata.CloudMetadata != nil {
+				server.SetCloudMetadata(metadata.CloudMetadata)
```

Comment:
Sow we're fetching metadata the first time by kicking off a background goroutine, but not actually setting the metadata until the next heartbeat, even if it's available much sooner.

Would it be simpler to just do:

```
go func() {
    metadata, err := fetch(...)
    server.SetCloudMetadata(metadata)
}()
```

(With a sync.Once and proper error handling, of course.)